test/Passes/DefaultPipeline/linalg-matmul-variants.mlir

```diff
@@ -0,0 +1,95 @@
+// RUN: tpp-opt %s -default-tpp-passes -split-input-file | FileCheck %s
+
+func.func @matmul(%arg0: tensor<2048x2048xbf16>, %arg1: tensor<2048x2048xbf16>, %arg2: tensor<2048x2048xbf16>)
+    -> tensor<2048x2048xbf16> {
+  %0 = linalg.matmul ins(%arg0, %arg1: tensor<2048x2048xbf16>, tensor<2048x2048xbf16>)
+                     outs(%arg2: tensor<2048x2048xbf16>)
+    -> tensor<2048x2048xbf16>
+  return %0 : tensor<2048x2048xbf16>
+}
+
+// CHECK-LABEL: @matmul(
+// CHECK-SAME: %[[ARG0:[a-zA-Z0-9]+]]: memref<2048x2048xbf16>,
+// CHECK-SAME: %[[ARG1:[a-zA-Z0-9]+]]: memref<2048x2048xbf16>,
+// CHECK-SAME: %[[ARG2:[a-zA-Z0-9]+]]: memref<2048x2048xbf16>
+// CHECK: memref.subview %[[ARG0]]
+// CHECK: call @xsmm_unary_invoke
+// CHECK: memref.subview %[[ARG1]]
+// CHECK: call @xsmm_unary_invoke
+// CHECK: memref.subview %[[ARG2]]
+// CHECK: call @xsmm_intel_amx_tile_config_invoke
```

Comment:
This should only work in SPR machines

Reply:
Do we have any logic that controls when these are generated?
Or is it just noop on other platforms?

Reply:
We do have a check for `x86` vs `arm` using libxsmm's CPUID, but that assumes host arch == target arch, so we didn't abuse of it. It should be fine for now, but we need to fix this once this logic goes to graph compilers.